### PR TITLE
Default to `React.StrictMode`

### PR DESCRIPTION
**Pull request by kyledurand**
I believe we want strict mode by default but I could be mistaken

### CODE REVIEW

File: polaris-react/.storybook/preview.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ function AppProviderDecorator(Story, context) {
 export const globalTypes = {
   strictMode: {
     name: 'React.StrictMode',
-    defaultValue: 'false',
+    defaultValue: 'true',
```
Comment (alex-page) [inline polaris-react/.storybook/preview.js:47]: I think we want to match `shopify/web`, if they use strict by default this would be fine to merge in.

**Review by alex-page — APPROVED** [review]

File: polaris-react/.storybook/preview.js (hunk 1 above)
Comment (BPScott) [reply polaris-react/.storybook/preview.js:47]: StrictMode has zero effect in production, it's just a development tool used to [check for correctness](https://reactjs.org/docs/strict-mode.html). I'd argue that helping ensure correct behaviour when developing in polaris-react is worthwhile even if web does not do that currently in its local development. Having polaris be strict mode compliant would make it easier for web and other consuming apps to adopt strict mode in their own codebases.

Comment (kyledurand) [reply polaris-react/.storybook/preview.js:47]: It doesn't look like they do which seems weird. We've always defaulted to strict mode in Polaris though. @BPScott any thoughts on this? Maybe we enable strict mode for storybook web as well?

Comment (BPScott) [reply polaris-react/.storybook/preview.js:47]: Oop sorry, that fell through the cracks.

Enabling strict mode in web's storybook sounds like a good idea to me! (also in their jest test harness)

**Pull request merged**